src/CommandAndStateMessageParser.cpp

```diff
@@ -258,8 +258,7 @@ string CommandAndStateMessageParser::parseDriveRevolutionCommandMessage(
     auto root = message["payload"]["devices"][address];
 
     auto vertical_cmd_sign = command.z() / abs(command.z());
-    auto vertical_cmd =
-        (abs(command.z()) + vertical_command_offset) * vertical_cmd_sign;
+    auto vertical_cmd = abs(command.z()) * vertical_cmd_sign + vertical_command_offset;
```

Comment:
Given that `command.z() = -0.1` and `vertical_command_offset = 0.2`. This would result in:
`0.1 * -1 + 0.2`. So a command to descend would turn into a command to ascend.

Reply:
Also `abs(x)` * `sign(x)` is the same as `x` anyways. And I dont know how I let this through, but if `command.z() = 0` the `vertical_cmd_sign` would be a floating point error. Do a check if its 0 before getting the sign.

Reply:
And add a specific test for that

Reply:
Actually I tested it out and the div by 0 returns nan, still not what we wanted. I guess the best option is to do something akin to the spaceship operator in ruby: if command.z() >= 0, sign = 1, else sign = -1.